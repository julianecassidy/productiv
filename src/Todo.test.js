import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const TEST_TODO = {
  title: "test todo",
  description: "test desc",
  priority: 1,
  id: 1,
};

describe("Todo component", function () {
  it("renders without crashing", function () {
    render(<Todo
      id={TEST_TODO.id} 
      title={TEST_TODO.title} 
      description={TEST_TODO.description} 
      priority={TEST_TODO.priority} />);
  });

  it("matches snapshot", function () {
    const { container } = render(
      <Todo 
        id={TEST_TODO.id} 
        title={TEST_TODO.title} 
        description={TEST_TODO.description} 
        priority={TEST_TODO.priority} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Shows the correct Todo on the page", function () {
    const { container } = render(<Todo
        id={TEST_TODO.id} 
        title={TEST_TODO.title} 
        description={TEST_TODO.description} 
        priority={TEST_TODO.priority} />)

    expect(container).toContainHTML("test desc");
    expect(container).toContainHTML("Priority: 1");
    expect(container).toContainHTML("test todo");
  });

});