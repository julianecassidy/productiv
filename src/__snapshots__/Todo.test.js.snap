// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        test todo
      </b>
       
      <small>
        Priority: 
        1
      </small>
    </div>
    <div>
      <small>
        test desc
      </small>
    </div>
  </div>
</div>
`;
